File: 04-SDD 实战项目/sdd-01-workitem-fullstack/workitem-backend/src/main/java/com/zmz/sdd/fullstack/app/domain/model/Workitem.java
package com.zmz.sdd.fullstack.app.domain.model;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * [SDD-SPEC: 02-功能规范.md §2.1]
 * 领域模型;不包含 deleted 字段(MP 逻辑删除字段仅 Entity 可见)
 */
@Data
@Builder
public class Workitem {
    private Long id;
    private String title;
    private String description;
    private WorkitemStatus status;
    private WorkitemPriority priority;
    private LocalDate dueDate;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: 04-SDD 实战项目/sdd-01-workitem-fullstack/workitem-frontend/src/types/workitem.ts
// [SDD-SPEC: 02-功能规范.md §2.1 ~ §2.3 + §1.1 Result + §3.2 PageVO]

export type WorkitemStatus = 'TODO' | 'DOING' | 'DONE'
export type WorkitemPriority = 'LOW' | 'MEDIUM' | 'HIGH'

export interface WorkitemVO {
  id: number
  title: string
  description: string | null
  status: WorkitemStatus
  priority: WorkitemPriority
  dueDate: string | null
  createdAt: string
  updatedAt: string
}

export interface WorkitemCreateReq {
  title: string
  description?: string
  priority?: WorkitemPriority
  dueDate?: string
}

export interface WorkitemUpdateReq {
  title?: string
  description?: string
  status?: WorkitemStatus
  priority?: WorkitemPriority
  dueDate?: string
}

export interface PageVO<T> {
  page: number
  size: number
  total: number
  totalPages: number
  list: T[]
}

export interface Result<T> {
  code: number
  message: string
  data: T | null
  timestamp: number
}
